pkg/kongstate/kongstate.go

```diff
@@ -98,7 +100,7 @@ func (ks *KongState) FillConsumersAndCredentials(log logrus.FieldLogger, s store
 				log.Errorf("failed to provision credential: empty secret")
 				continue
 			}
-			err = c.SetCredential(credType, credConfig)
+			err = c.SetCredential(credType, credConfig, ks.Version)
```

Comment:
Codecode points out this line isn't covered by tests, though I understand that it already existed that way prior. Since we changed the method signature and functionality can we add some test of this to `kongstate_test.go`?

Reply:
7297568 is not great but at least covers this. Surprised that we didn't have any existing parser tests that hit this.

Testing this a bit annoying since the change to state happens in place and because KongState isn't conducive to comparisons. `FillConsumersAndCredentials()` adds empty arrays for _everything_ it touches, so if you try to populate just what you want in the want state, comparisons fail with something like https://gist.github.com/rainest/7328fb6f9089b0548ecbcc2e04607d68

Not sure there's any good way to design test arrays, as we normally want, in light of that.

A single test that checks specific fields only at least confirms that the function doesn't break spectacularly and does do something, so :shrug:

Reply:
As long as we can create some issue that tracks our need to round up tests for these existing code paths (or we comment on an existing issue that covers that) I don't feel this needs to block this PR :+1: